daphne_worker/src/dap.rs

```diff
@@ -553,12 +553,20 @@ where
             .iter()
             .zip(agg_store_responses.into_iter())
         {
+            let min_time = self.least_valid_report_time();
+            let max_time = self.greatest_valid_report_time();
             for metadata in span.get(bucket).unwrap() {
                 let processed = reports_processed.contains(&metadata.id);
                 if processed && !collected {
                     early_fails.insert(metadata.id.clone(), TransitionFailure::ReportReplayed);
-                } else if !processed && collected {
+                } else if !processed && collected || metadata.time < min_time {
+                    // We don't have a "too late" code, so we just say its batch has been
+                    // collected like we do with the !processed and collected case.  We don't
+                    // want to say ReportDropped, as that would imply we couldn't tell if it
+                    // was valid or not, but we know it is not valid.
```

Comment:
Does that me we need a TooLate code in the std?

Reply:
We might consider adding this, yeah. Unfortunately in DAP-03 we're restricted to the values enumerated in https://www.ietf.org/archive/id/draft-ietf-ppm-dap-03.html#name-aggregate-initialization. In the meantime, I would call this "report_dropped".

Reply:
We just got rid of the equivalent `reportTooLate` problem type in https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/333, so I'm skeptical that it's needed here. The question is: what is the leader going to do with `reportTooLate` that they wouldn't do if they saw `reportDropped`? Either way, all the leader can do is just not include this particular report in the next round of the aggregate protocol.

Reply:
I'm ok either way, I just was worried that "report dropped" had additional meaning to it, in particular that the leader might decide that the batch was uncollectable because it thought we had lost the ability to check validity.  I'll go with dropped for now.